pkgs/top-level/all-packages.nix

```diff
@@ -36922,11 +36922,9 @@ with pkgs;
 
   principia = callPackage ../games/principia { };
 
-  prismlauncher-unwrapped = kdePackages.callPackage ../games/prismlauncher {
-    inherit (darwin.apple_sdk.frameworks) Cocoa;
-  };
+  prismlauncher-unwrapped = kdePackages.callPackage ../by-name/pr/prismlauncher-unwrapped/package.nix { };
 
-  prismlauncher = kdePackages.callPackage ../games/prismlauncher/wrapper.nix { };
+  prismlauncher = kdePackages.callPackage ../by-name/pr/prismlauncher/package.nix { };
```

Comment:
by-name hierarchy does not work with customized callPackage.

This issue was revisited many times before:

https://github.com/NixOS/nixpkgs/pull/305509#issuecomment-2099663683